### PR TITLE
fix: Fix bugs in new webserver

### LINKED ISSUE #919: Refactor web server
It would be nice to hide all the implementation of the web server into a library and make usage simple. Usage should be like:
```cpp
Server server(ip, port);
server.onGet("/", getCallback);
serveer.onPost("/", postCallback);
server.run();
```

### PR DESCRIPTION
**Pull request by kuznetsss**
Fixes #919.

Fixes bugs for new webserver:
- Unhandled exception when closing already closed websocket
- No pings for plain websocket connection
- Server drops websocket connection when client responds to pings but doesn't send anything

Also changing API of ng connections. Now timeout is set by a separate method instead of providing it for each call.

### CODE REVIEW

File: src/web/ng/Connection.hpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ class Connection : public ConnectionMetadata {
     /**
      * @brief The default timeout for send, receive, and close operations.
      */
-    static constexpr std::chrono::steady_clock::duration DEFAULT_TIMEOUT = std::chrono::seconds{30};
+    static constexpr std::chrono::steady_clock::duration DEFAULT_TIMEOUT = std::chrono::seconds{10};
```
Comment (cindyyan317) [inline src/web/ng/Connection.hpp]: I don't see this change in PR summary. Is it typo? or intended?

**Review by cindyyan317 — DISMISSED** [review]
LGTM .
Left a question about the default value.

File: src/web/ng/Connection.hpp (hunk 1 above)
Comment (kuznetsss) [reply src/web/ng/Connection.hpp]: This is intended. I thought it is better to disconnect inactive clients earlier.

**Review by godexsoft — APPROVED** [review]
LGTM 👍

**Pull request merged**